src/Uno.UI.Tests/Windows_UI_Xaml/Given_ResourceDictionary.cs

```diff
@@ -279,5 +311,273 @@ public void When_Has_Custom_Theme()
+		}
+
+		[TestMethod]
+		public void When_Enumerated()
+		{
+			var rd = new ResourceDictionary();
+			rd["foot"] = "Boot";
+			var inner = new ResourceDictionary();
+			inner["hand"] = "Glove";
+			rd.MergedDictionaries.Add(inner);
+			var def = new ResourceDictionary();
+			def["eyes"] = "Sunglasses";
+			rd.ThemeDictionaries["Default"] = def;
+
+			var keys = new List<object>();
+			foreach (var kvp in rd)
+			{
+				keys.Add(kvp.Key);
+			}
+
+			Assert.IsTrue(keys.Contains("foot"));
+
+			Assert.IsFalse(keys.Contains("hand"));
+			Assert.IsTrue(rd.ContainsKey("hand"));
+
+			Assert.IsFalse(keys.Contains("eyes"));
+			Assert.IsTrue(rd.ContainsKey("eyes"));
+		}
+
+		[TestMethod]
+		public void When_Implicit_Style_From_Code()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var control = new Test_Control();
+			Assert.IsTrue(control.Resources.ContainsKey(typeof(RadioButton)));
+			Assert.IsNotNull(control.Resources[typeof(RadioButton)] as Style);
+		}
+
+		[TestMethod]
+		public void When_Merged_Dictionary_Local()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			Assert.IsFalse(app.Resources.ContainsKey("NotInAppResources")); // These dictionaries shouldn't end up in App.Resources, or this test won't test anything
+
+			var control = new Test_Control();
+			control.TestTextBlock2.DataContext = true;
+			AssertEx.AssertHasColor(control.TestTextBlock2.Foreground, Colors.Orange);
+		}
+
+		[TestMethod]
+		public void When_Merged_Dictionary_Style()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			Assert.IsFalse(app.Resources.ContainsKey("NotInAppResources")); // These dictionaries shouldn't end up in App.Resources, or this test won't test anything
+
+			var page = new Test_Page();
+
+			app.HostView.Children.Add(page);
+
+			page.Measure(new Size(1000, 1000));
+
+			AssertEx.AssertHasColor(page.TestProgressRing.Foreground, Colors.Fuchsia);
+		}
+
+		[TestMethod]
+		public void When_Local_Dictionary_Intra_Reference()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var control = new Test_Control();
+			control.TestTextBlock2.DataContext = false;
+			AssertEx.AssertHasColor(control.TestTextBlock2.Foreground, Colors.LimeGreen);
+		}
+
+		[TestMethod]
+		public void When_Local_Lexically_Forward()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var page = new Test_Page();
+			page.TestTextBlock2.DataContext = true;
+
+			AssertEx.AssertHasColor(page.TestTextBlock2.Foreground, Colors.MidnightBlue);
+		}
+
+		[TestMethod]
+		public void When_Direct_Local_Assignation_From_Source()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var page = new Test_Page();
+
+			AssertEx.AssertContainsColorBrushResource(page.TestStackPanel.Resources, "NotInAppResources", Colors.Cyan);
+		}
+
+		[TestMethod]
+		public void When_Created_From_Source_In_Codebehind()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'app'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4826539)